processor/spanmetricsprocessor/processor_test.go

```diff
@@ -333,10 +373,13 @@ func TestProcessorConsumeTraces(t *testing.T) {
 			mexp := &mocks.MetricsExporter{}
 			tcon := &mocks.TracesConsumer{}
 
-			//TODO: CLAIRE: CHECK IF WE CAN ADD TO THESE TESTS
 			// Mocked metric exporter will perform validation on metrics, during p.ConsumeTraces()
 			mexp.On("ConsumeMetrics", mock.Anything, mock.MatchedBy(func(input pdata.Metrics) bool {
-				return tc.verifier(t, input, false, make(map[string]int))
+				if tc.verifier != nil {
```

Comment:
Why would it be ok for the verifier to be `nil`? This is what validates the metrics. 

Also, even if you don't need the verifier for a test case, I would probably just set it as an empty function `func(...) {}` in the test case, rather than checking if it's nil for every test case.

Reply:
The main purpose of the test was to check that it didn't cause an out of bounds index error rather than verifying the correct metrics were generated so that's why I set it to nil. 

I guess it couldn't hurt to verify anyway but the existing metric verifier also wasn't written generically enough so that I could easily use it for the specific use case and I didn't think it was particularly worth rewriting given that we are planning on removing the aggregation which will simplify things and our existing tests will still give us enough confidence and testing.

I'll do the suggestion for setting the empty function 👍